### PR TITLE
Add avoidance plugin

### PR DESCRIPTION
**Pull request by coderkalyan**
Proto file for the avoidance plugin, a continuation of https://github.com/mavlink/MAVSDK/pull/974

### CODE REVIEW

**Review by julianoes — COMMENTED** [review]
Nice, could you add some description to the methods on when to send them? Like do I have to publish the waypoint trajectory and/or the bezier trajectory? Is that a choice?

**Comment by coderkalyan** [comment]
> Nice, could you add some description to the methods on when to send them? Like do I have to publish the waypoint trajectory and/or the bezier trajectory? Is that a choice?

Sure thing, I just mocked this up a couple days ago, happy to document further. The waypoint versus bezier trajectory is an either/or thing AFAIK, I've been reading the PX4 avoidance docs and thats what it says.

File: protos/avoidance/avoidance.proto [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    /*
+     * Publish a bezier trajectory to the autopilot.
+     */
+    rpc PublishBezierTrajectory(PublishBezierTrajectoryRequest) returns (PublishBezierTrajectoryResponse) {}
+}
+
+// Position of waypoint
+message Position {
+    float x = 1; // X component of position
```
[the remark this answers is not in the document]
Comment (coderkalyan) [reply protos/avoidance/avoidance.proto:28]: Will do.

File: protos/avoidance/avoidance.proto [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    Position position = 2; // Position at waypoint
+    Velocity velocity = 3; // Velocity at waypoint
+    Acceleration acceleration = 4; // Acceleration at waypoint
+    float yaw_angle = 5; // Yaw angle
+    float yaw_rate = 6; // Yaw rate
+}
+
+// Bezier node, makes up a trajectory
+message Bezier {
+    int64 timestamp = 1; // Timestamp of bezier node
```
[the remark this answers is not in the document]
Comment (coderkalyan) [reply protos/avoidance/avoidance.proto:59]: Same here, will do.

**Comment by coderkalyan** [comment]
@julianoes Updated with your comments, but I'm keeping this as WIP right now since I'm still developing the backend implementation and theres a good chance I'll need to add some other things.

**Review by julianoes — APPROVED** [review]
Looking good.

**Comment by JonasVautherin** [comment]
Out of curiosity: any progress on the C++ implementation, @coderkalyan?

**Comment by coderkalyan** [comment]
@JonasVautherin Unfortunately not, my school work has been quite high lately - but I do intend on completing this, hopefully in the next week or 2. :smiley: Apologies for the extreme delay...

**Pull request closed without merging**